File: app/src/androidTest/java/com/example/task2/ui/benchmark/TestMapBenchmark.java
package com.example.task2.ui.benchmark;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;

import androidx.test.ext.junit.runners.AndroidJUnit4;

import com.example.task2.R;
import com.example.task2.models.benchmarks.CellOperation;
import com.example.task2.models.benchmarks.MapBenchmark;

import org.junit.runner.RunWith;

import java.util.List;

@RunWith(AndroidJUnit4.class)
public class TestMapBenchmark extends TestBenchmark {

    @Override
    protected List<CellOperation> createItemsList() {
        return new MapBenchmark().createItemsList(false);
    }

    @Override
    public void setUp() {
        super.setUp();
        onView(withText(R.string.tab_maps)).perform(click());
    }
}

File: app/src/androidTest/java/com/example/task2/ui/benchmark/TestMapBenchmark.kt
package com.example.task2.ui.benchmark

import androidx.test.espresso.Espresso
import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.ext.junit.runners.AndroidJUnit4
import com.example.task2.R
import com.example.task2.models.benchmarks.CellOperation
import com.example.task2.models.benchmarks.MapBenchmark
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class TestMapBenchmark : TestBenchmark() {
    override fun createItemsList(): List<CellOperation> = MapBenchmark().createItemsList(false)

    override fun setUp() {
        super.setUp()
        Espresso.onView(ViewMatchers.withText(R.string.tab_maps)).perform(ViewActions.click())
    }
}
